tools/tpm2_nvwrite.c

```diff
@@ -61,13 +62,16 @@ struct tpm_nvwrite_ctx {
     UINT16 offset;
     char *raw_pcrs_file;
     tpm2_session *policy_session;
+    tpm2_session *hmac_session;
+    bool hmac_auth;
```

Comment:
I don't think we need these. You should extend [tpm2_auth_util_from_optarg](https://github.com/tpm2-software/tpm2-tools/blob/master/lib/tpm2_auth_util.c#L112) to support the hmac keyword. Store the type of session in the session *, and smarten up the rest of the session logic to avoid calling session save on hmac sessions.

Reply:
@williamcroberts there is one problem with this approach. If hmac-auth was the only authentication session the suggested approach would work. The way tpm2_auth_util_from_optarg is it takes only one session as an input. For example if i have a policy session requiring hmac as the only form of authentication then I would need to do something like this:
tpm2_nvwrite -P session:HEX -P hmac:PASSWORD
Any thoughts?

Reply:
So you're saying you need both the sessionHandle and the hmac data in the TPMS_AUTH_COMMAND data?
Right now that is not supported.

You could do the `-P` approach as you outlined. But the tool would need to be smart enough to complain if you try to set the same field twice, ie:
* 2+ passwords
* 2+ hmac's,
* 2+ sessions
* 1 password and 1 hmac 

You could also use a comma to separate them:
`-P session:<file>,hmac:<password>`

I think I like the multiple -P option approach that you outline. I would make the tpm2_auth_ routine the one smart enough to check if it's set before setting it, so we don't have to duplicate that logic everywhere.

Reply:
Implemented